src/mlpack/tests/callback_test.cpp

```diff
@@ -257,4 +259,19 @@ BOOST_AUTO_TEST_CASE(RBMCallbackTest)
   BOOST_REQUIRE_GT(stream.str().length(), 0);
 }
 
+BOOST_AUTO_TEST_CASE(SparseAutoencodeCallbackTest)
+{
+  // Simple fake dataset.
+  arma::mat data1("0.1 0.2 0.3 0.4 0.5;"
+                  "0.1 0.2 0.3 0.4 0.5;"
+                  "0.1 0.2 0.3 0.4 0.5;"
+                  "0.1 0.2 0.3 0.4 0.5;"
+                  "0.1 0.2 0.3 0.4 0.5");
+
+  ens::L_BFGS optimizer(5, 100);
+  ens::StoreBestCoordinates<arma::mat> cb;
+  mlpack::nn::SparseAutoencoder encoder2(data1, 2, 1, 0, 0, 0 , optimizer, cb);
```

Comment:
```suggestion
  mlpack::nn::SparseAutoencoder encoder2(data1, 5, 1, 0, 0, 0 , optimizer, cb);
```

I think that should work.

Reply:
Nice find , thanks :)

Reply:
Haven't tested it, but let's see.